src/Tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema.UnitTests/SchemaGenerationTests.cs

```diff
@@ -77,31 +78,43 @@ private async Task<Stream> GenerateSchema()
             return new FileStream(tempSchema, FileMode.Open, FileAccess.Read, FileShare.ReadWrite, 4096, FileOptions.DeleteOnClose);
         }
 
-        private async Task<bool> CompareLines(TextReader first, TextReader second)
+        private async Task<bool> CompareLines(TextReader baseline, TextReader generated)
         {
-            IList<string> firstLines = await ReadAllLines(first);
-            IList<string> secondLines = await ReadAllLines(second);
+            IList<string> baselineLines = await ReadAllLines(baseline);
+            IList<string> generatedLines = await ReadAllLines(generated);
 
-            for (int i = 0; i < Math.Min(firstLines.Count, secondLines.Count); i++)
+            for (int i = 0; i < Math.Min(baselineLines.Count, generatedLines.Count); i++)
             {
-                if (!string.Equals(firstLines[i], secondLines[i], StringComparison.Ordinal))
+                if (!string.Equals(baselineLines[i], generatedLines[i], StringComparison.Ordinal))
                 {
                     _outputHelper.WriteLine($"Differs from baseline on line {i + 1}:");
-                    _outputHelper.WriteLine(firstLines[i]);
-                    _outputHelper.WriteLine(secondLines[i]);
+                    PrintSection(_outputHelper, nameof(baseline), baselineLines, i);
+                    PrintSection(_outputHelper, nameof(generated), generatedLines, i);
                     return false;
                 }
             }
 
-            if (firstLines.Count != secondLines.Count)
+            if (baselineLines.Count != generatedLines.Count)
             {
-                _outputHelper.WriteLine($"Count differs from baseline: {firstLines.Count} {secondLines.Count}");
+                _outputHelper.WriteLine($"Count differs from baseline: {baselineLines.Count} {generatedLines.Count}");
                 return false;
             }
 
             return true;
         }
 
+        private static void PrintSection(ITestOutputHelper outputHelper, string header, IList<string> lines, int lineHighlighted, int contextQty = 7)
+        {
+            outputHelper.WriteLine($"-----{header}-----");
+            int startLine = Math.Max(0, lineHighlighted - contextQty);
+            int endLine = Math.Min(lines.Count, lineHighlighted + contextQty);
+            int formatQty = (endLine + 1).ToString("D").Length; // Get the length of the biggest number (add 1 for the 1-based index)
+            for (int i = startLine; i <= endLine; i++)
+            {
+                outputHelper.WriteLine("{0}:{1}{2}", (i+1).ToString("D" + formatQty, CultureInfo.InvariantCulture), (i == lineHighlighted) ? " >" : "  ", lines[i]);
```

Comment:
This was my actual suggestion, maybe not conveyed correctly:
```suggestion
                outputHelper.WriteLine("{0}:{1}{2}", (i+1).ToString("D" + formatQty.ToString(CultureInfo.InvariantCulture)), (i == lineHighlighted) ? " >" : "  ", lines[i]);
```